nomad/structs/structs.go

```diff
@@ -8099,22 +8201,15 @@ func (wc *WaitConfig) Copy() *WaitConfig {
 }
 
 func (wc *WaitConfig) Equal(o *WaitConfig) bool {
-	if wc.Min == nil && o.Min != nil {
-		return false
+	if wc == nil || o == nil {
+		return wc == o
 	}
-
-	if wc.Max == nil && o.Max != nil {
-		return false
-	}
-
-	if wc.Min != nil && (o.Min == nil || *wc.Min != *o.Min) {
+	switch {
+	case !pointer.Eq(wc.Min, o.Min):
```

Comment:
I don't think this is actually equivalent to the code it's replacing, because of how it handles the `wc.Min == nil && o.Min != nil` case. But I suspect that means we have a bug in the existing code? https://go.dev/play/p/lgiv0IVVLD3

Reply:
I believe these are equivelent; I've updated `TestTaskWaitConfig_Equals` with the negative assertions missing covering these cases. I then swap the old/new implementations and the tests continue to pass.

In your playground, the third case would never happen because it's guarded by:
https://github.com/hashicorp/nomad/blob/v1.5.0/nomad/structs/structs.go#L8093

Reply:
Oops, missed that. LGTM